django/core/management/commands/dumpdata.py

```diff
@@ -163,13 +163,16 @@ def get_objects(count_only=False):
                         "%s is a proxy model and won't be serialized." % model._meta.label,
                         category=ProxyModelWarning,
                     )
-                if not model._meta.proxy and router.allow_migrate_model(using, model):
+                _using = using
+                if _using is None:
+                    _using = router.db_for_read(model)
```

Comment:
I would change
```diff --git a/django/core/management/commands/dumpdata.py b/django/core/management/commands/dumpdata.py
index df9eecc0f8..953f261ef4 100644
--- a/django/core/management/commands/dumpdata.py
+++ b/django/core/management/commands/dumpdata.py
@@ -67,7 +67,7 @@ class Command(BaseCommand):
     def handle(self, *app_labels, **options):
         format = options['format']
         indent = options['indent']
-        using = options['database']
+        database = options['database']
         excludes = options['exclude']
         output = options['output']
         show_traceback = options['traceback']
```
to simplify this.

We should also skip models with different `db` for read, instead of checking `router.allow_migrate_model()`, e.g.
```python
using = router.db_for_read(model)
if not model._meta.proxy and (not database or using == database):
    if use_base_manager:
        ...
```